src/roster/views.py

```diff
@@ -1490,11 +1500,26 @@ def get(self, request, *args, **kwargs):
             self.clusive_student = None
 
         self.roster = period.users.exclude(user=request.user, role=Roles.TEACHER).order_by('user__first_name')
+
+        # Get the reading data for the current student. This data will be shared by all panels on the student details page
+        reading_data = Paradata.reading_books_for_period_student(self.clusive_user.current_period, days=self.days, sort='name', username=kwargs['username'])[0]
```

Comment:
Minor: I suggest using `.first()` instead of `[0]`, especially if the set can be empty.  The `.first()` function will return `None` if it is empty, whereas `[0]` will raise an exception.

See:  https://docs.djangoproject.com/en/3.2/ref/models/querysets/#first

Reply:
Ignore my previous comment.  I thought that the return type was a`QuerySet` but it's a list, and `first()` won't work with that.

Reply:
Your concern about no reading data being returned makes a lot of sense. I added the logic to handle this case.

Reply:
LGTM